pkg/execution/state/redis_state/queue.go

```diff
@@ -3541,6 +3558,132 @@ func (q *queue) InProgress(ctx context.Context, prefix string, concurrencyKey st
 	return q.primaryQueueShard.RedisClient.unshardedRc.Do(ctx, cmd).AsInt64()
 }
 
+func (q *queue) IteratePartitions(ctx context.Context) error {
+	l := logger.StdlibLogger(ctx)
+
+	// Check on global partition and queue partition sizes
+	var offset, total int64
+	chunkSize := q.partitionIteratorChunkSize
+
+	// Semaphore to limit backlog enrollment concurrency
+	sem := semaphore.NewWeighted(q.backlogEnrollConcurrency)
+
+	// Wait until all partitions have been processed
+	wg := sync.WaitGroup{}
+
+	r := q.primaryQueueShard.RedisClient.unshardedRc
+	// iterate through all the partitions in the global partitions in chunks
+	for {
+		// grab the global partition by chunks
+		cmd := r.B().Zrange().
+			Key(q.primaryQueueShard.RedisClient.kg.GlobalPartitionIndex()).
+			Min("-inf").
+			Max("+inf").
+			Byscore().
+			Limit(offset, chunkSize).
+			Build()
+
+		pkeys, err := r.Do(ctx, cmd).AsStrSlice()
+		if err != nil {
+			return fmt.Errorf("error retrieving partitions for instrumentation: %w", err)
+		}
```

Comment:
might be cool to make this a `PartitionIterator` with an option to ignore leases or something.

it'll be nice to reduce these sort of direct redis command usage over the codebase, so it's more concentrated and make the wrapper more flexible.

don't have to do this in a PR, so would be good to at least document it as a follow up.